### PR TITLE
Fixes VSTS Bug 767140: 7.7 has regressed in typing performance

### PR DESCRIPTION
**Pull request by mkrueger**
https://devdiv.visualstudio.com/DevDiv/_workitems/edit/767140

### CODE REVIEW

**Comment by mkrueger** [comment]
@monojenkins backport release-7.8

**Comment by mkrueger** [comment]
Fold segment has been updated to tasks, some left overs from the background thread days are just obsolete. This patch should improve typing performance a bit as well.

**Comment by mkrueger** [comment]
@monojenkins backport release-7.8

**Comment by mkrueger** [comment]
@monojenkins backport release-7.8

**Comment by mkrueger** [comment]
shouldn't use atom to edit files :)

**Comment by mkrueger** [comment]
@monojenkins backport release-7.8

File: main/src/core/Mono.TextEditor.Shared/Mono.TextEditor/Document/TextDocument.cs [hunk 1]
```diff
@@ -1394,33 +1390,25 @@ HashSet<FoldSegment> UpdateFoldSegmentWorker (IEnumerable<IFoldSegment> segments
...
 			}
 			bool countChanged = foldedSegments.Count != newFoldedSegments.Count;
-			update = foldedSegmentAdded || countChanged;
+			update = foldedSegmentAdded || countChanged || foldedFoldingRemoved;
 			return newFoldedSegments;
 		}
-		
+
+		[Obsolete("Doesn't do anything anymore")]
 		public void WaitForFoldUpdateFinished ()
```
Comment (Therzok) [inline main/src/core/Mono.TextEditor.Shared/Mono.TextEditor/Document/TextDocument.cs]: We can break API in master.

**Pull request merged**